imports/wasi_snapshot_preview1/fs_test.go

```diff
@@ -1172,9 +1190,60 @@ func Test_fdReaddir(t *testing.T) {
 				}
 			},
 			buf: 0, bufLen: 30, // length is longer than the second entry
-			cookie:          2,  // previous d_next
-			expectedBufused: 30, // length to read exactly one more.
-			expectedMem:     append(dirent2, dirent3[0:4]...),
+			cookie:          1,  // previous d_next
+			expectedBufused: 30, // length to read some more, but not enough for a header.
+			expectedMem:     append(dirent2),
+			expectedMemSize: len(dirent2), // we do not want to compare the full buffer since we don't know what the leftover 4 bytes will contain.
+			expectedReadDir: &internalsys.ReadDir{
+				CountRead: 3,
+				Entries:   testDirEntries[1:],
+			},
+		},
+		{
+			name: "read second and another header",
```

Comment:
I have added this check to validate the logic when there is space for 1 record but not enough for a second but it does have enough for the header of the second.